### PR TITLE
Feature/m2 1095 respondents api

### PR DESCRIPTION
**Pull request by kdanilenka32**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "i18next": "^22.0.4",
     "i18next-browser-languagedetector": "^7.0.0",
     "javascript-time-ago": "^2.5.9",
+    "lodash.debounce": "^4.0.8",
     "lodash.uniqueid": "^4.0.1",
     "lodash.debounce": "4.0.8",
```
Comment (muhurov) [inline package.json:29]: is it necessary to have duplicate?

File: package.json [hunk 2]
```diff
@@ -73,6 +74,7 @@
     ]
   },
   "devDependencies": {
+    "@types/lodash.debounce": "^4.0.7",
     "@types/lodash.uniqueid": "^4.0.7",
     "@types/lodash.debounce": "4.0.7",
```
Comment (muhurov) [inline package.json:79]: is it necessary to have duplicate?

File: src/index.tsx [hunk 3]
```diff
@@ -13,7 +13,7 @@ document.body.appendChild(svgBuilder(svgSprite));
 const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement);
 root.render(
   <React.StrictMode>
-    <App />
+    <App />,
```
Comment (muhurov) [inline src/index.tsx:16]: Why is comma here?

File: src/shared/components/Search/Search.tsx [hunk 4]
```diff
@@ -8,7 +9,7 @@ import { SearchProps } from './Search.types';
 export const Search = ({ onSearch, ...props }: SearchProps) => (
   <StyledTextField
     {...props}
-    onChange={(event: ChangeEvent<HTMLInputElement>) => onSearch(event.target.value)}
+    onChange={debounce((event: ChangeEvent<HTMLInputElement>) => onSearch(event.target.value), 700)}
```
Comment (muhurov) [inline src/shared/components/Search/Search.tsx:12]: please move 700 to const

Comment (kdanilenka32) [reply src/shared/components/Search/Search.tsx]: fixed

File: package.json (hunk 2 above)
Comment (kdanilenka32) [reply package.json]: fixed

File: package.json (hunk 1 above)
Comment (kdanilenka32) [reply package.json]: fixed

**Review by muhurov — APPROVED** [review]

File: src/modules/Dashboard/api/api.ts
Comment (IhorNesterenko99) [inline src/modules/Dashboard/api/api.ts:1]: Could we get rid of the unneeded API function getAccountUserListApi which took users from the old API?

File: src/modules/Dashboard/components/Table/Table.tsx
Comment (IhorNesterenko99) [inline src/modules/Dashboard/components/Table/Table.tsx:1]: Maybe rename to DashboardTable?

File: src/shared/layouts/BaseLayout/BaseLayout.tsx [hunk 5]
```diff
@@ -18,22 +18,42 @@ export const BaseLayout = () => {
...
   useEffect(() => {
     if (accountData?.account && isAuthorized) {
       dispatch(folders.thunk.getAppletsForFolders({ account: accountData.account }));
     }
   }, [dispatch, accountData?.account, isAuthorized]);
 
+  useEffect(() => {
+    const { getWorkspaceApplets } = applets.thunk;
+    const { getWorkspaceUsers } = users.thunk;
+
```
Comment (IhorNesterenko99) [inline src/shared/layouts/BaseLayout/BaseLayout.tsx:34]: const ownerId = currentWorkspaceData?.ownerId;
and pls use this variable to get rid of repeated code

File: src/shared/layouts/BaseLayout/BaseLayout.tsx [hunk 6]
```diff
@@ -7,9 +7,9 @@ import {
   TransferOwnershipPopup,
 } from 'modules/Dashboard/features/Applet/Popups';
 import { useAppDispatch } from 'redux/store';
-import { auth } from 'modules/Auth/state';
-import { account, folders, popups, users } from 'modules/Dashboard/state';
-import { Footer } from 'shared/components';
+import { account, folders, popups, workspaces, applets, users, auth } from 'redux/modules';
+import { DEFAULT_ROWS_PER_PAGE, Footer } from '../..//components';
```
Comment (IhorNesterenko99) [inline src/shared/layouts/BaseLayout/BaseLayout.tsx:11]: could you pls use full routes for '../..//components' and  '../../types'?

File: src/modules/Dashboard/components/Table/Table.types.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { Dispatch, SetStateAction, MouseEvent } from 'react';
+import { Row, UiType } from 'shared/components';
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/components/Table/Table.types.ts:2]: pls add correct spacing to imports

File: src/modules/Dashboard/features/Applets/Table/Table.types.ts [hunk 8]
```diff
@@ -1,20 +1,8 @@
-import { Dispatch, SetStateAction } from 'react';
-
 import { Applet, FolderApplet } from 'redux/modules';
-import { HeadCell, Order } from 'shared/types/table';
 
-import { OrderBy } from '../Applets.types';
+import { TableProps as SharedTableProps } from '../../../components';
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/Applets/Table/Table.types.ts:3]: could you pls change this: '../../../components' ?

File: src/modules/Dashboard/features/Respondents/Respondents.tsx [hunk 9]
```diff
@@ -25,15 +22,19 @@ import {
   RespondentsRemoveAccessPopup,
   EditRespondentPopup,
 } from './Popups';
+import { Table } from '../../components';
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/Respondents/Respondents.tsx:25]: could you pls change this: '../../components'?

File: src/modules/Dashboard/features/Respondents/Respondents.utils.tsx [hunk 10]
```diff
@@ -48,7 +48,7 @@ export const getActions = ({
 ];
 
 export const getChosenAppletData = (
-  respondentsItems: Users,
+  respondentsItems: any,
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/Respondents/Respondents.utils.tsx:51]: pls don't use any type

File: src/modules/Dashboard/state/Users/Users.thunk.ts [hunk 11]
```diff
@@ -3,24 +3,19 @@ import { AxiosError } from 'axios';
 
 import { ApiError } from 'redux/modules';
 
-import { getAccountUserListApi, updatePinApi, UpdatePin, getInvitationsApi } from 'api';
+import {
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/state/Users/Users.thunk.ts:6]: pls add correct spacing

File: src/modules/Dashboard/features/Respondents/Respondents.utils.tsx (hunk 10 above)
Comment (kdanilenka32) [reply src/modules/Dashboard/features/Respondents/Respondents.utils.tsx:51]: will be fixed when strart integration

File: src/shared/layouts/BaseLayout/BaseLayout.tsx (hunk 6 above)
Comment (kdanilenka32) [reply src/shared/layouts/BaseLayout/BaseLayout.tsx]: let's keep it, to avoid circular dep

File: src/modules/Dashboard/components/Table/Table.tsx
Comment (kdanilenka32) [reply src/modules/Dashboard/components/Table/Table.tsx:1]: need something more informative or keep it

File: src/shared/layouts/BaseLayout/BaseLayout.tsx (hunk 6 above)
Comment (IhorNesterenko99) [reply src/shared/layouts/BaseLayout/BaseLayout.tsx]: as I think circular diapedesis occur after importing from one directory, for example shared/components
if we have different: shared/components and shared/layouts there are no circular deps

File: src/modules/Dashboard/features/Applet/AddUser/AddUserForm/AddUserForm.tsx [hunk 12]
```diff
@@ -15,7 +15,6 @@ import {
 import { StyledErrorText } from 'shared/styles/styledComponents';
 import { getAppletInvitationApi } from 'api';
 import { getErrorMessage } from 'shared/utils/errors';
-import { prepareUsersData } from 'shared/utils/prepareUsersData';
 import { setAccountName } from 'modules/Auth/state/Auth.thunk';
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/Applet/AddUser/AddUserForm/AddUserForm.tsx:18]: import from modules/auth/state

File: src/shared/components/Search/Search.tsx [hunk 13]
```diff
@@ -1,14 +1,19 @@
 import { ChangeEvent } from 'react';
+import debounce from 'lodash.debounce';
 
 import { Svg } from 'shared/components/Svg';
```
Comment (IhorNesterenko99) [inline src/shared/components/Search/Search.tsx:5]: pls add correct spacing

File: src/shared/layouts/BaseLayout/components/LeftBar/LeftBar.tsx [hunk 14]
```diff
@@ -5,20 +5,22 @@ import { useTranslation } from 'react-i18next';
 
 import { StyledLabelMedium, variables } from 'shared/styles';
 import { SwitchWorkspace, WorkspaceImage } from 'shared/features/SwitchWorkspace';
-import { Workspace } from 'shared/features/SwitchWorkspace/SwitchWorkspace.types';
 import { getWorkspacesApi } from 'shared/api';
-import { auth } from 'modules/Auth/state';
+
+import { workspaces as currentWorkspace, auth, Workspace } from 'redux/modules';
 import { useAsync } from 'shared/hooks';
```
Comment (IhorNesterenko99) [inline src/shared/layouts/BaseLayout/components/LeftBar/LeftBar.tsx:12]: pls add correct spacings

File: src/modules/Dashboard/features/Applet/AddUser/AddUserForm/AddUserForm.tsx (hunk 12 above)
Comment (kdanilenka32) [reply src/modules/Dashboard/features/Applet/AddUser/AddUserForm/AddUserForm.tsx:18]: it wiil be deleteted

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**